lib/api.js

```diff
@@ -113,11 +146,38 @@ function makeApiRequest(transport, opts, bodyObj, callback) {
       respData = respData.join('');
       parseApiResponse(respData, callback);
     });
+
+    dequeueApiRequest(req);
   });
 
+  pendingQueue.push(req);
+
+  var originalArguments = arguments;
```

Comment:
We are still using the full `arguments` object here, again not sure how big of a deal this is.